src/backends/policy_enforcement/souffle/value_test.cc

```diff
@@ -0,0 +1,172 @@
+  int64_t num = GetParam();
+  Number number_value = Number(num);
+  std::string datalog_str = number_value.ToDatalogString();
+  int64_t parsed_int = 0;
+  bool conversion_succeeds = absl::SimpleAtoi(datalog_str, &parsed_int);
+  ASSERT_TRUE(conversion_succeeds);
+  ASSERT_EQ(parsed_int, num);
+}
+
+static int64_t kSampleIntegerValues[] = {0, -1, 1,
+                                         std::numeric_limits<long>::max(),
+                                         std::numeric_limits<long>::min()};
+
+INSTANTIATE_TEST_SUITE_P(NumberTest, NumberTest,
+                         ValuesIn(kSampleIntegerValues));
+
+class SymbolTest : public TestWithParam<absl::string_view> {};
+
+TEST_P(SymbolTest, SymbolTest) {
+  absl::string_view symbol = GetParam();
+  Symbol symbol_value = Symbol(symbol);
+  std::string symbol_str = symbol_value.ToDatalogString();
+  ASSERT_EQ(symbol_str, "\"" + std::string(symbol) + "\"");
+}
+
+static absl::string_view kSampleSymbols[] = {"", "x", "foo", "hello_world"};
+
+INSTANTIATE_TEST_SUITE_P(SymbolTest, SymbolTest, ValuesIn(kSampleSymbols));
+
+using SimpleRecord = Record<Symbol, Number>;
+
+TEST(SimpleRecordNilTest, SimpleRecordNilTest) {
+  ASSERT_EQ(Record<SimpleRecord>().ToDatalogString(), "nil");
+}
+
+class SimpleRecordTest
+    : public TestWithParam<std::tuple<absl::string_view, int64_t>> {};
+
+TEST_P(SimpleRecordTest, SimpleRecordTest) {
+  const auto [symbol, number] = GetParam();
+  SimpleRecord record_value = SimpleRecord(Symbol(symbol), Number(number));
+  ASSERT_EQ(record_value.ToDatalogString(),
+            absl::StrFormat(R"(["%s", %d])", symbol, number));
+}
+
+INSTANTIATE_TEST_SUITE_P(SimpleRecordTest, SimpleRecordTest,
+                         Combine(ValuesIn(kSampleSymbols),
+                                 ValuesIn(kSampleIntegerValues)));
+
+class NumList : public Record<Number, NumList> {
+ public:
+  using Record::Record;
+};
+
+struct NumListAndExpectedDatalog {
+  const NumList *num_list_ptr;
+  absl::string_view expected_datalog;
+};
+
+class NumListTest : public TestWithParam<NumListAndExpectedDatalog> {};
+
+TEST_P(NumListTest, NumListTest) {
+  const auto [num_list_ptr, expected_datalog] = GetParam();
+  EXPECT_EQ(num_list_ptr->ToDatalogString(), expected_datalog);
+}
+
+static const NumList kEmptyNumList;
+static const NumList kOneElementNumList = NumList(Number(5), NumList());
+static const NumList kTwoElementNumList(Number(-30),
+                                        NumList(Number(28), NumList()));
+
+static NumListAndExpectedDatalog kListAndExpectedDatalog[] = {
+    {.num_list_ptr = &kEmptyNumList, .expected_datalog = "nil"},
+    {.num_list_ptr = &kOneElementNumList, .expected_datalog = "[5, nil]"},
+    {.num_list_ptr = &kTwoElementNumList,
+     .expected_datalog = "[-30, [28, nil]]"}};
+
+INSTANTIATE_TEST_SUITE_P(NumListTest, NumListTest,
+                         ValuesIn(kListAndExpectedDatalog));
+
+static constexpr char kArithAdtName[] = "Arith";
```

Comment:
Better not to abbreviate. Call it `Arithmetic` instead of `Arith` here and elsewhere?

Reply:
Fixed.